### ISSUE TITLE
DLRS 2.17 Insufficient 

### ISSUE THREAD

**Issue by techsavy7** [SFDO-Community/declarative-lookup-rollup-summaries#1249]
**IMPORTANT NOTE** The Chatter Community Group can be found [here](https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p). If it is an issue that you need help on configuring the tool or have an unexpected error its really worth seeing if the community can help first before raising a GitHub issue. Most users hangout in the Chatter Community Group. Otherwise if prefer you can post here and still link in the Chatter Community Group.

We have installed DLRS 2.17 is our prototype org and encountered the following:

Unable to connect to the Salesforce Metadata API.
Web service callout failed: WebService returned a SOAP Fault: INSUFFICIENT_ACCESS: Access from current IP address is not allowed faultcode=sf:INSUFFICIENT_ACCESS faultactor=

Gone through forums and feed back for this issue.

Ensured the following :

1. Unchecked "Lock sessions to IP address from which they originated"
2. 
3. Also the DLRS home page in the org, points out an Important Post install step and when the "Create/Update Remote Site setting" button is clicked  it fails with a message
DEBUG USE ONLY: Web Service callout failed: webService returned a SOAP Fault: INSUFFICIENT_ACCESS: Access from Current IP address is not allowed faultcode=sf:INSUFFICIENT_ACCESS faultactor=

Can you help resolve the issue, why is this needed and what is the internal callout that requires this access to Salesforce Metadata API

**Comment by stephenbrown1** [comment]
This could be based on login IP ranges that you have defined in the org. Have you checked the login audit trail to see what the IP is? It's probably from a SFDC IP address which is calling the metadata API to get further info on the org. You may need to add it to the network IP list - if login IP ranges are used, then any other IP (incl Salesforce ones makig callouts) are denied access.
https://help.salesforce.com/s/articleView?id=000333031&type=1

**Comment by techsavy7** [comment]
Thanks @stephenbrown1. 

Was able to identify the IPs from the audit trail and they resolve to US West 2 Amazon AWS compute ec2 instances and they are not a part of the published Salesforce trusted IPs.

Why would these IPs whitelisted and if they are not trusted SF IP ranges. can you share some additional insights.

**Comment by stephenbrown1** [comment]
agreed, the callout from the welcome page just checks the metadata API, using Url.getOrgDomainUrl().toExternalForm(), which Salesforce says shouldn't need explicit remote site setting access. It does set up a separate session to make that call, and uses the sessionId from the existing session. Have you recently enabled a custom domain (ie <name>.my.salesforce.com)?
could you also check whether this is also checked?
_Select Require HttpOnly attribute to restrict session ID cookie access. A cookie with the HttpOnly attribute isn’t accessible through non-HTTP methods, such as calls from JavaScript._
(in session settings)

**Comment by techsavy7** [comment]
1. we do have our custom domain config set up and 
2. Require HttpOnly attribute is checked

**Comment by techsavy7** [comment]
@aheber, this is showing unchecked in our org. I have gone through some of the previous solutions offered in the forum and ensured the same. Thanks

**Comment by stephenbrcnz** [comment]
In my dev org where it works ok, all of the 6 options in session settings are unchecked. Incl the httponly option. Can you try this please and see if it works ok?

**Comment by techsavy7** [comment]
Thanks for your response. the error still persists

**Comment by techsavy7** [comment]
Also the IPs that need to be whitelisted, are these the IPs of the Salesforce Org where the package is installed?

**Comment by stephenbrcnz** [comment]
Ok - if you can whitelist the IP that you see from the login audit trail, in the network access - trusted IP range, and let me know what that IP is also, will check it out.. thanks

**Comment by techsavy7** [comment]
The IPs seem to be the same. Unfortunately, am unable to share them in the public forum. 
Company policy now prevents usage of non-appexchange apps. 

Thank you for your time and support
